packages/react-client/src/ReactFlightClient.js

```diff
@@ -266,7 +266,11 @@ ReactPromise.prototype.then = function <T>(
       initializeModuleChunk(chunk);
       break;
   }
-  if (__DEV__ && enableAsyncDebugInfo) {
+  if (
+    __DEV__ &&
+    enableAsyncDebugInfo &&
+    (typeof resolve !== 'function' || !(resolve: any).isReactInternalListener)
```

Comment:
The microtask here means that cycles don't resolve synchronously which means that console logs can miss them. This is not the only thing that can be async though. If you console log a client reference that hasn't loaded yet you get the same problem.

In the future, we should really make sure that the debug info can be parsed async.